python/tests/rd_tests/test_sum.py

```diff
@@ -756,6 +734,49 @@ def test_resample_extrapolate(self):
                     rd_sum.get_interp_direct(key_rate, t),
                 )
 
+    def test_pandas2_compatibility_dataframe_index(self):
+        # regression test to verify that pandas frames in pandas 2
+        # does not break due to missing collection for time_index
+        path = os.path.join(self.TESTDATA_ROOT, "local/ECLIPSE/cp_simple3/SHORT.UNSMRY")
+        smry = Summary(path)
+        try:
+            smry.pandas_frame(
+                time_index=smry.time_range(interval="1Y"), column_keys=["WELL:NAME"]
+            )
+        except TypeError as err:
+            pytest.fail(repr(err))
+
+
+def create_time_vector(lst):
+    vec = TimeVector()
+    for l in lst:
+        vec.append(l)
+    return vec
+
+
+@pytest.mark.parametrize("time_index_type", [list, create_time_vector, tuple])
+def test_pandas(time_index_type):
```

Comment:
Is it possible to pick a better name for this test?

Reply:
Let's go for `test_summary_to_pandas_frame` as that at is at least unambiguous. The test is a bit unspecific about what it requires of the function so writing a good `test_that_...` name seems a bit optimistic.

Reply:
done